sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -584,36 +583,39 @@ def upload_blob(
             client=self._client,
             **kwargs)
         if blob_type == BlobType.BlockBlob:
-            return upload_block_blob(**options)
+            return upload_block_blob(**options)  # type: ignore
         if blob_type == BlobType.PageBlob:
-            return upload_page_blob(**options)
-        return upload_append_blob(**options)
+            return upload_page_blob(**options)  # type: ignore
+        return upload_append_blob(**options)  # type: ignore
```

Comment:
These 3 you should be able to get right by properly typing the return of the `upload_*_blob` functions.

Reply:
Yep. Ouch. Will fix this shortly when I revisit that PR and then propagate the changes up 😢

Reply:
Used the `cast` pattern here and fixed, will be in the next propagation...